rpm/src/main/java/org/eclipse/packager/rpm/deps/Dependencies.java

```diff
@@ -113,36 +112,21 @@ public static List<Dependency> getEnhances(final ReadableHeader<RpmTag> header)
     private static List<Dependency> getDependencies(final ReadableHeader<RpmTag> header, final RpmTag namesTag, final RpmTag versionsTag, final RpmTag flagsTag) {
         Objects.requireNonNull(header);
 
-        final Object rawNames = header.getValue(namesTag).orElse(null);
-        final Object rawVersions = header.getValue(versionsTag).orElse(null);
-        Object rawFlags = header.getValue(flagsTag).orElse(null);
-
-        if (rawFlags instanceof Integer[]) {
-            final Integer[] iflags = (Integer[]) rawFlags;
-            final int[] flags = new int[iflags.length];
-            for (int i = 0; i < iflags.length; i++) {
-                flags[i] = iflags[i];
-            }
-            rawFlags = flags;
-        }
-
-        if (rawNames instanceof String[] && rawVersions instanceof String[] && rawFlags instanceof int[]) {
-            final String[] names = (String[]) rawNames;
-            final String[] versions = (String[]) rawVersions;
-            final int[] flags = (int[]) rawFlags;
-
-            if (names.length == versions.length && names.length == flags.length) {
-                final List<Dependency> result = new ArrayList<>(names.length);
-                for (int i = 0; i < names.length; i++) {
-                    final String name = names[i];
-                    final String version = versions[i];
-                    final Set<RpmDependencyFlags> flagSet = RpmDependencyFlags.parse(flags[i]);
-                    result.add(new Dependency(name, version, flagSet));
-                }
-                return result;
+        final List<String> names = header.getStringList(namesTag);
```

Comment:
I think we make some breaking changes on how this function behaves. Which is ok. However, I also think we should document that.

So far, to my understanding, the function simply ignored unexpected values and types. Now it will throw an exception. And while it's a private method, I think it makese sense adding a note and tracing calls that lead to this call and put a note on the API.

Reply:
Is the current javadoc acceptable? I put it in the main text now as `@throws` because it's unchecked, but I wasn't sure about that.